flang/include/flang/Optimizer/Dialect/FIROps.td

```diff
@@ -2138,9 +2138,11 @@ def fir_DoLoopOp : region_Op<"do_loop",
     Index:$lowerBound,
     Index:$upperBound,
     Index:$step,
+    Variadic<AnyType>:$reduceOperands,
     Variadic<AnyType>:$initArgs,
```

Comment:
```suggestion
    Variadic<AnyType>:$initArgs,
    Variadic<AnyType>:$reduceOperands,
```

initArgs is more related to the control operands so maybe it's better to keep them together.